### PR TITLE
[ssu] Make _ca-certificate optional. Fixes JB#60436

### PR DESCRIPTION
**Pull request by keto**

### CODE REVIEW

File: libssu/ssu.cpp [hunk 1]
```diff
@@ -684,7 +677,15 @@ void Ssu::updateCredentials(bool force)
...
     QList<QSslCertificate> caCertificates;
-    caCertificates << QSslCertificate::fromPath(ssuCaCertificate);
+    if (ssuCaCertificate.isEmpty()) {
+        // When the localCertificate is set, it seems that the system CA
+        // certificates are not used by default and need to be added explicitly
+        // (if we are not using a custom CA certificate).
+        // Could not track down why this happens exactly.
+        caCertificates << sslConfiguration.systemCaCertificates();
+    } else {
+        caCertificates << QSslCertificate::fromPath(ssuCaCertificate);
```
Comment (abranson) [inline libssu/ssu.cpp:687]: Should we add the system certs in both cases? I don't know if defining a cert in ssu should replace all of the system certs. I'd expect it to be used in addition to them.

Comment (keto) [reply libssu/ssu.cpp:687]: I think it is supposed to use only the defined certificate if it is defined, kind of certificate pinning.

Comment (abranson) [reply libssu/ssu.cpp:687]: Ok, it's fine either way I think. Should that comment read 'When the localCertificate *isn't* set' instead though?

**Review by abranson — APPROVED** [review]

File: libssu/ssu.cpp (hunk 1 above)
Comment (keto) [reply libssu/ssu.cpp:687]: The comment is correct. When locaCertificate, i.e the client certificate, is set, then it does not load the system CA certificates automatically. In comparison the registration request, which does not use client certificate, works fine without setting the system CA certificates. I'll try to clarify the comment.

**Pull request merged**